packages/nextcloud/test/webdav_test.dart

```diff
@@ -19,17 +19,78 @@ void main() {
       final baseURL = Uri.parse(values.$1);
       final sanitizedBaseURL = Uri.parse(values.$2);
 
-      test('$baseURL', () {
-        expect(WebDavClient.constructUri(baseURL).toString(), '$sanitizedBaseURL$webdavBase');
-        expect(WebDavClient.constructUri(baseURL, Uri(path: '/')).toString(), '$sanitizedBaseURL$webdavBase');
-        expect(WebDavClient.constructUri(baseURL, Uri(path: 'test')).toString(), '$sanitizedBaseURL$webdavBase/test');
-        expect(WebDavClient.constructUri(baseURL, Uri(path: 'test/')).toString(), '$sanitizedBaseURL$webdavBase/test');
-        expect(WebDavClient.constructUri(baseURL, Uri(path: '/test')).toString(), '$sanitizedBaseURL$webdavBase/test');
-        expect(WebDavClient.constructUri(baseURL, Uri(path: '/test/')).toString(), '$sanitizedBaseURL$webdavBase/test');
+      test(baseURL, () {
+        expect(
+          WebDavClient.constructUri(baseURL).toString(),
+          '$sanitizedBaseURL$webdavBase',
+        );
+        expect(
+          WebDavClient.constructUri(baseURL, PathUri(path: '/')).toString(),
+          '$sanitizedBaseURL$webdavBase',
+        );
+        expect(
+          WebDavClient.constructUri(baseURL, PathUri(path: 'test')).toString(),
+          '$sanitizedBaseURL$webdavBase/test',
+        );
+        expect(
+          WebDavClient.constructUri(baseURL, PathUri(path: 'test/')).toString(),
+          '$sanitizedBaseURL$webdavBase/test',
+        );
+        expect(
+          WebDavClient.constructUri(baseURL, PathUri(path: '/test')).toString(),
+          '$sanitizedBaseURL$webdavBase/test',
+        );
+        expect(
+          WebDavClient.constructUri(baseURL, PathUri(path: '/test/')).toString(),
+          '$sanitizedBaseURL$webdavBase/test',
+        );
       });
     }
   });
 
+  group('PathUri', () {
+    test('path', () {
```

Comment:
I think that the path of an empty `PathUri()` should be `/`.

Reply:
I think `.` which is an empty string here is more intuitive.

Reply:
Then please document this behavior in the `PathUri`.